### ISSUE TITLE
Uncaught TypeError: Cannot read property 'getChildPos' of null

### ISSUE THREAD

**Issue by studiorooster** [matjaztrontelj/pinegrow-atom#1]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 1.8.0
**System**: Mac OS X 10.11.5
**Thrown From**: [pinegrow-atom](https://github.com/matjaztrontelj/pinegrow-atom) package, v0.3.1
### Stack Trace

Uncaught TypeError: Cannot read property 'getChildPos' of null

```
At remote_modules/pinegrowparser.js:870

TypeError: Cannot read property 'getChildPos' of null
    at [object Object].pgParserNode.next (remote_modules/pinegrowparser.js:870:26)
    at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1580:21)
    at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1584:31)
    at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1584:31)
    at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1584:31)
    at /Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:248:46
    at doWithCurrentSourceNodeForEditor (/Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:21:7)
    at /Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:245:17
    at Array.forEach (native)
    at Socket.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:243:17)
    at Socket.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-emitter/index.js:131:20)
    at Socket.onevent (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/socket.js:263:10)
    at Socket.onpacket (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/socket.js:221:12)
    at Manager.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-bind/index.js:21:15)
    at Manager.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-emitter/index.js:131:20)
    at Manager.ondecoded (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/manager.js:333:8)
    at Decoder.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-bind/index.js:21:15)
    at Decoder.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/socket.io-parser/node_modules/component-emitter/index.js:134:20)
    at Decoder.add (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/socket.io-parser/index.js:247:12)
    at Manager.ondata (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/manager.js:323:16)
    at Socket.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-bind/index.js:21:15)
    at Socket.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/component-emitter/index.js:134:20)
    at Socket.onPacket (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/socket.js:441:14)
    at WS.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/socket.js:258:10)
    at WS.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/component-emitter/index.js:134:20)
    at WS.Transport.onPacket (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/transport.js:143:8)
    at WS.Transport.onData (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/transport.js:135:8)
    at WebSocket.ws.onmessage (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/transports/websocket.js:132:10)
```
### Commands

```
```
### Config

``` json
{
  "core": {},
  "pinegrow-atom": {
    "pinegrowUrl": "http://127.0.0.1:40000"
  }
}
```
### Installed Packages

``` coffee
# User
atom-django, v0.3.2
autocomplete-python, v1.8.1
language-gitignore, v0.2.0
pigments, v0.29.2
pinegrow-atom, v0.3.1
python-autopep8, v0.1.3

# Dev
No dev packages
```

**Comment by matjaztrontelj** [comment]
Hi Charles,

can you please send me the copy of HTML code to support@pinegrow.com where
this issue happens?

Thanks, Matjaz

On Mon, Jun 20, 2016 at 4:08 PM, Charles Smith notifications@github.com
wrote:

> [Enter steps to reproduce below:]
> 1. ...
> 2. ...
> 
> _Atom Version_: 1.8.0
> _System_: Mac OS X 10.11.5
> _Thrown From_: pinegrow-atom
> https://github.com/matjaztrontelj/pinegrow-atom package, v0.3.1
> Stack Trace
> 
> Uncaught TypeError: Cannot read property 'getChildPos' of null
> 
> At remote_modules/pinegrowparser.js:870
> 
> TypeError: Cannot read property 'getChildPos' of null
>     at [object Object].pgParserNode.next (remote_modules/pinegrowparser.js:870:26)
>     at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1580:21)
>     at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1584:31)
>     at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1584:31)
>     at [object Object].pgParserNode.getPositionInSource (remote_modules/pinegrowparser.js:1584:31)
>     at /Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:248:46
>     at doWithCurrentSourceNodeForEditor (/Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:21:7)
>     at /Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:245:17
>     at Array.forEach (native)
>     at Socket.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/lib/pinegrow-atom.js:243:17)
>     at Socket.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-emitter/index.js:131:20)
>     at Socket.onevent (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/socket.js:263:10)
>     at Socket.onpacket (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/socket.js:221:12)
>     at Manager.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-bind/index.js:21:15)
>     at Manager.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-emitter/index.js:131:20)
>     at Manager.ondecoded (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/manager.js:333:8)
>     at Decoder.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-bind/index.js:21:15)
>     at Decoder.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/socket.io-parser/node_modules/component-emitter/index.js:134:20)
>     at Decoder.add (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/socket.io-parser/index.js:247:12)
>     at Manager.ondata (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/lib/manager.js:323:16)
>     at Socket.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/component-bind/index.js:21:15)
>     at Socket.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/component-emitter/index.js:134:20)
>     at Socket.onPacket (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/socket.js:441:14)
>     at WS.<anonymous> (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/socket.js:258:10)
>     at WS.Emitter.emit (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/node_modules/component-emitter/index.js:134:20)
>     at WS.Transport.onPacket (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/transport.js:143:8)
>     at WS.Transport.onData (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/transport.js:135:8)
>     at WebSocket.ws.onmessage (/Users/rooster/.atom/packages/pinegrow-atom/node_modules/socket.io-client/node_modules/engine.io-client/lib/transports/websocket.js:132:10)
> 
> Commands
> 
> Config
> 
> {
>   "core": {},
>   "pinegrow-atom": {
>     "pinegrowUrl": "http://127.0.0.1:40000"
>   }
> }
> 
> Installed Packages
> 
> # User
> 
> atom-django, v0.3.2
> autocomplete-python, v1.8.1
> language-gitignore, v0.2.0
> pigments, v0.29.2
> pinegrow-atom, v0.3.1
> python-autopep8, v0.1.3
> 
> # DevNo dev packages
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> https://github.com/matjaztrontelj/pinegrow-atom/issues/1, or mute the
> thread
> https://github.com/notifications/unsubscribe/AABQ9-gf1eqrP_AqpT1gr-EG57HPzaolks5qNp7sgaJpZM4I5vCQ
> .

## 

_Matjaz Trontelj_
Author of Pinegrow Web Editor http://pinegrow.com/

**Comment by studiorooster** [comment]
It happened when I ran Atom Beautify

**Comment by matjaztrontelj** [comment]
Will be fixed in Pinegrow 2.92. Thanks!
